youtube_dl/extractor/jwplatform.py

```diff
@@ -50,7 +50,11 @@ def _parse_jwplayer_data(self, jwplayer_data, video_id=None, require_title=True,
             # JWPlayer backward compatibility: flattened sources
             # https://github.com/jwplayer/jwplayer/blob/v7.4.3/src/js/playlist/item.js#L29-L35
             if 'sources' not in video_data:
-                video_data['sources'] = [video_data]
+                if isinstance(video_data, list):
```

Comment:
Specifically, here's what we get without that hunk of the patch; it's not possible that this change breaks other websites since they would already generate this TypeError:
```
pb3:youtube-dl jhawk$ python test/test_download.py TestDownload.test_IQM2_1
[IQM2] 1679: Downloading webpage
[IQM2] 1563: Downloading webpage
E
======================================================================
ERROR: test_IQM2_1 (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 141, in test_template
    force_generic_extractor=params.get('force_generic_extractor', False))
  File "/Users/jhawk/src/youtube-dl/youtube_dl/YoutubeDL.py", line 694, in extract_info
    ie_result = ie.extract(url)
  File "/Users/jhawk/src/youtube-dl/youtube_dl/extractor/common.py", line 355, in extract
    return self._real_extract(url)
  File "/Users/jhawk/src/youtube-dl/youtube_dl/extractor/iqm2.py", line 88, in _real_extract
    webpage, video_id, require_title=False)
  File "/Users/jhawk/src/youtube-dl/youtube_dl/extractor/jwplatform.py", line 33, in _extract_jwplayer_data
    jwplayer_data, video_id, *args, **kwargs)
  File "/Users/jhawk/src/youtube-dl/youtube_dl/extractor/jwplatform.py", line 57, in _parse_jwplayer_data
    video_data['sources'] = [video_data]
TypeError: list indices must be integers, not unicode

----------------------------------------------------------------------
Ran 1 test in 2.128s

FAILED (errors=1)
```

And here's a little context via pdb:
```
pb3:youtube-dl jhawk$ python  -c 'import pdb, youtube_dl; print youtube_dl; pdb.runcall(youtube_dl.main, ["-v","http://cambridgema.iqm2.com/Citizens/SplitView.aspx?Mode=Video&MeetingID=1679"])' youtube_dl 
<module 'youtube_dl' from 'youtube_dl/__init__.pyc'>
> /Users/jhawk/src/youtube-dl/youtube_dl/__init__.py(444)main()
-> try:
(Pdb) b youtube_dl/extractor/jwplatform.py:57
Breakpoint 1 at /Users/jhawk/src/youtube-dl/youtube_dl/extractor/jwplatform.py:57
(Pdb) c
...
[IQM2] 1563: Downloading webpage
> /Users/jhawk/src/youtube-dl/youtube_dl/extractor/jwplatform.py(57)_parse_jwplayer_data()
-> video_data['sources'] = [video_data]
(Pdb) p video_data
[{u'default': False, u'tracks': [{u'default': True, u'kind': u'captions', u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt', u'label': u'English'}], u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_480.mp4', u'label': u'SD 480'}, {u'default': True, u'tracks': [{u'default': True, u'kind': u'captions', u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt', u'label': u'English'}], u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_720.mp4', u'label': u'HD 720'}]
(Pdb) p video_data['sources']
*** TypeError: TypeError('list indices must be integers, not str',)
(Pdb) p jwplayer_data
{u'playlist': [[{u'default': False, u'tracks': [{u'default': True, u'kind': u'captions', u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt', u'label': u'English'}], u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_480.mp4', u'label': u'SD 480'}, {u'default': True, u'tracks': [{u'default': True, u'kind': u'captions', u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt', u'label': u'English'}], u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_720.mp4', u'label': u'HD 720'}]]}
(Pdb) pp jwplayer_data
{u'playlist': [[{u'default': False,
                 u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_480.mp4',
                 u'label': u'SD 480',
                 u'tracks': [{u'default': True,
                              u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt',
                              u'kind': u'captions',
                              u'label': u'English'}]},
                {u'default': True,
                 u'file': u'http://MediaHTTP.IQM2.com/CambridgeMA/1563_720.mp4',
                 u'label': u'HD 720',
                 u'tracks': [{u'default': True,
                              u'file': u'/Services/TranscriptGet.aspx?MediaID=1563&format=vtt',
                              u'kind': u'captions',
                              u'label': u'English'}]}]]}
(Pdb) 
```